### PR TITLE
Console over websocket

### PR DESCRIPTION
**Pull request by jukkar**
This is implementation of zephyr console over a websocket. Tested with Firefox and wsdump.py command line tool which are able to send and receive data to/from the console.
There are generic websocket support patches in this PR so that compilation will work, only last two patches are related to ws-console implementation.

### CODE REVIEW

**Review by dbkinder — APPROVED** [review]

File: samples/net/ws_console/README.rst [hunk 1]
```diff
@@ -0,0 +1,35 @@
+.. _websocket-server-sample:
```
Comment (dbkinder) [inline samples/net/ws_console/README.rst]: whoops, just notice in https://app.shippable.com/github/zephyrproject-rtos/zephyr/runs/3894/3/tests that this document label is already in use (must be unique across all the documents)

**Review by dbkinder — CHANGES_REQUESTED** [review]
doc label conflict

**Comment by pfalcon** [comment]
@andrewboie : Adding you here. You asked about integrating printk into console subsystem lately. But there're bigger ("better") integration issues. So, we have subsys/console which supports only UART, but uses interrupt mode, and thus known to work. It also has POSIXy/ANSICy design. And we have drivers/console, which has adhoc design, known to be lossy (I recently tried to use bluetooth shell, which uses drivers/console/uart, and I couldn't use it, because it requires typing in long commands, and loses characters on clipboard paste from the host). But there now will be whole 3 of those! But they're really drivers - an app needs to configure (call) explicitly console drivers which it wants to work with.

So, a good task would be integrating those "console drivers" into "console subsystem". That would require substantial rewrite of those drivers (telnet/websocket), to work with generic POSIX/ANSIC model instead of Zephyr's adhoc one.

Hope this comment is useful.

File: subsys/shell/shell.c [hunk 2]
```diff
@@ -570,6 +573,9 @@ void shell_init(const char *str)
 #ifdef CONFIG_TELNET_CONSOLE
 	telnet_register_input(&avail_queue, &cmds_queue, completion);
 #endif
+#ifdef CONFIG_WS_CONSOLE
+	ws_register_input(&avail_queue, &cmds_queue, completion);
```
Comment (pfalcon) [inline subsys/shell/shell.c]: @andrewboie : Here's what I mean by "an app needs to call each console driver explicitly".

**Review by tbursztyka — APPROVED** [review]
seems fine to me

**Comment by jukkar** [comment]
Rebased over #1271 and made sure this works ok.

**Comment by jukkar** [comment]
Updated to latest websocket code + Cmake changes.

Note that this PR contains some patches that are also found in #1271 
This is not to be merged to v1.10

**Comment by jukkar** [comment]
This is already merged as part of other PR.

**Pull request closed without merging**